packages/oats/src/generate-types.ts

```diff
@@ -145,6 +149,9 @@ export function run(options: Options) {
       return;
     }
     if (additional === true || additional == null) {
+      if (options.unknownAdditionalPropertiesIndexSignature === AdditionalPropertiesIndexSignature.omit) {
```

Comment:
if the flag is on do not emit the index signature for true or omitted additionalproperties